### PR TITLE
lucide-static

### PR DESCRIPTION
**Pull request by ericfennis**
# Lucide Static

This package include the following lucide implementations:

- All svg files
- Javascript library containing strings of svgs. (Node.js)
- Icon fonts
- Svg sprite

Closes: #313

### CODE REVIEW

**Comment by siarie** [comment]
Hi @ericfennis,
may i know if this PR is related to [this issue](https://github.com/lucide-icons/lucide/issues/392)?
I see you mention *icon fonts* here.

**Comment by ericfennis** [comment]
@siarie Sorry for my late reply.
Yes this npm package also contains icon fonts, so this will be fixed partially.

File: packages/lucide-static/scripts/generateSprite.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    name: 'symbol',
+    type: 'element',
+    attributes: {
+      id: name,
+      viewBox: '0 0 24 24',
+      fill: 'none',
+      stroke: 'currentColor',
+      'stroke-width': '2',
+      'stroke-linecap': 'round',
+      'stroke-linejoin': 'round',
```
Comment (BinaryKitten) [inline packages/lucide-static/scripts/generateSprite.js:17]: Hi, Based on Discord conversation - having these on the symbol in the sprite prevents them from being properly set via CSS.
Not having them would mean that the end user would need to add them to their SVG import - either via CSS class or directly on the element.

File: packages/lucide-static/README.md [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+<!-- Icon Sprite, not recommended for production! -->
+<img src="https://unpkg.com/lucide-static@latest/sprite.svg#home">
+
+<!-- or -->
+<svg
+  xmlns="http://www.w3.org/2000/svg"
+  width="24"
+  height="24"
+  viewBox="0 0 24 24"
+>
```
Comment (BinaryKitten) [inline packages/lucide-static/README.md:85]:
```suggestion
  viewBox="0 0 24 24"
  fill="none"
  stroke="currentColor"
  stroke-width="2"
  stroke-linecap="round"
  stroke-linejoin="round"
>
```

File: packages/lucide-static/README.md [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+  <use
+    href="https://unpkg.com/lucide-static@latest/sprite.svg#alert-triangle"
+  />
+</svg>
+```
+
```
Comment (BinaryKitten) [inline packages/lucide-static/README.md:91]:
```suggestion

If you'd prefer, you can use CSS to hold your base SVG properties
```css
.lucide-icon {
    width: 24px;
    height: 24px;
    stroke: currentColor;
    fill: none;
    stroke-width: 2;
    stroke-linecap: round;
    stroke-linejoin: round;
}
    ```
and update the svg as follows
```svg
<svg
  xmlns="http://www.w3.org/2000/svg"
  class="lucide-icon"
>
<use
    href="https://unpkg.com/lucide-static@latest/sprite.svg#alert-triangle"
  />
</svg>
    ```
```

Apologies on the formatting on the suggestion here - the closing code markers need to be un-indented - but doing so broke Github's suggestion

**Review by BinaryKitten — COMMENTED** [review]
Suggestions with Documentation update

**Comment by ericfennis** [comment]
@BinaryKitten Thanks for you review and testing. I will test this.

**Comment by madeleineostoja** [comment]
Very much looking forward to this, for my use-case importing icon SVGs directly is far more ergonomic than running some script that injects SVG content for you. Any idea when this might get shipped, even as an rc?

**Comment by ericfennis** [comment]
@madeleineostoja thanks for letting me know. I think I'm ready to merge this. I needed to check if all the sprite implementations worked. And it is.

**Pull request merged**